app/assets/javascripts/admin/addon/components/bulk-user-delete-confirmation.gjs

```diff
@@ -0,0 +1,199 @@
+import Component from "@glimmer/component";
+import { tracked } from "@glimmer/tracking";
+import { on } from "@ember/modifier";
+import { action } from "@ember/object";
+import { service } from "@ember/service";
+import { TrackedArray } from "@ember-compat/tracked-built-ins";
+import DButton from "discourse/components/d-button";
+import DModal from "discourse/components/d-modal";
+import { ajax } from "discourse/lib/ajax";
+import { extractError } from "discourse/lib/ajax-error";
+import { bind } from "discourse-common/utils/decorators";
+import { i18n } from "discourse-i18n";
+
+const BULK_DELETE_CHANNEL = "/bulk-user-delete";
+
+export default class BulkUserDeleteConfirmation extends Component {
+  @service messageBus;
+
+  @tracked confirmButtonDisabled = true;
+  @tracked deleteStarted = false;
+  @tracked logs = new TrackedArray();
+  failedUsernames = [];
+
+  callAfterBulkDelete = false;
+
+  constructor() {
+    super(...arguments);
+    this.messageBus.subscribe(BULK_DELETE_CHANNEL, this.onDeleteProgress);
+  }
+
+  willDestroy() {
+    super.willDestroy(...arguments);
+    this.messageBus.unsubscribe(BULK_DELETE_CHANNEL, this.onDeleteProgress);
+  }
+
+  get confirmDeletePhrase() {
+    return i18n(
+      "admin.users.bulk_actions.delete.confirmation_modal.confirmation_phrase",
+      { count: this.args.model.userIds.length }
+    );
+  }
+
+  #logError(line) {
```

Comment:
I really like the logging UI here, I feel like it could be its own component. We do something like this already for backups and for site updates, is there any way these could be reused or for this one to supersede them in a fancy new component (in a separate PR of course)?

Reply:
Possibly. My main concern with this implementation would be performance — I don't know how well this implementation scales if used in a place where thousands of log entries could be printed. In this case, we're limited to 100 users so performance isn't a huge concern.

Reply:
Okay, I guess we just keep an eye out for more use cases, would be nice to have some unified logger component that is performant that we can use throughout the admin UI